### ISSUE TITLE
EBGAN for a pre-trained discriminator

### ISSUE THREAD

**Issue by KangOxford** [KangOxford/GAIL-Fail#2]
In the Energy Based GAN, discriminator is an auto-encoder, only trained with real data. Then we can firstly train this discriminator and keep it fixed.
[A video for it. Time: 1:02:30 - 1:04:30. Open in a new page](https://www.youtube.com/watch?v=dFwesaqC_Wo&list=WL&index=11&t=141s&ab_channel=Hung-yiLee)

<img width="860" alt="image" src="https://user-images.githubusercontent.com/37290277/172031432-c23a9699-1c46-46f2-bd09-e7210f15fe0a.png">

**Comment by KangOxford** [comment]
Or Boundary Equibilibrium Generative Adversarial Networks [BEGAN](https://github.com/artcg/BEGAN), which is an improved version of EBGAN.

**Comment by KangOxford** [comment]
same reward means same policy

**Comment by KangOxford** [comment]
format of energy softmax

**Comment by KangOxford** [comment]
Here when we are trying to apply ebgan, we must have a specifical distribution. This constraints on the specific policy means the eb-gail not easy to be widely applied

**Comment by KangOxford** [comment]
D2-imitation: if the policy is deterministic, then we can turn the discriminator into a binary-classifier. And in fact, a huge part of the problems are with deterministic policy.

**Comment by KangOxford** [comment]
We want to find that if the policy is undeterministic then how to better set the discriminator?
